assembl/static2/js/app/components/administration/languageMenu.jsx

```diff
@@ -10,47 +11,63 @@ import Ja from '../svg/flags/ja';
 
 // TO DO get it dynamically
 const Flag = ({ locale }) => {
-  switch (locale) {
-  case 'en':
-    return <En />;
-  case 'fr':
-    return <Fr />;
-  case 'ja':
-    return <Ja />;
-  default:
+
+  const localeOnly = locale ? locale.split("_")[0] : null ;
+  if (localeOnly) {
+    const name = capitalize(localeOnly);
+    const el = React.createElement(name, {}, null);
     return (
-      <span>
-        {locale}
-      </span>
+      <span>{el}</span>
     );
   }
+  return (<span></span>); //Best approach?
+
+  // switch (locale) {
+  // case 'en':
+  //   return <En />;
+  // case 'fr':
+  //   return <Fr />;
+  // case 'ja':
+  //   return <Ja />;
+  // default:
+  //   return (
+  //     <span>
+  //       {locale}
+  //     </span>
+  //   );
+  // }
 };
 
-const LanguageMenu = ({ changeLocale, selectedLocale, translations }) => {
-  return (
-    <div className="relative">
-      <div className="language-menu">
-        <OverlayTrigger placement="top" overlay={languageTooltip}>
-          <div>
-            {Object.keys(translations).map((key, index) => {
-              return (
-                <div
-                  onClick={() => {
-                    return changeLocale(key);
-                  }}
-                  id={key}
-                  className={selectedLocale === key ? 'flag-container active' : 'flag-container'}
-                  key={index}
-                >
-                  <Flag locale={key} />
-                </div>
-              );
-            })}
-          </div>
-        </OverlayTrigger>
+const LanguageMenu = ({ changeLocale, selectedLocale, translations, visibility }) => {
+  if (visibility) {
```

Comment:
use an HOC for visibility

Reply:
Do you mean to wrap this component in another that controls what it shows? I could certainly do it, but do you have a pro/con of HOC vs prop?